### ISSUE TITLE
Support permalinks configured in head matter

### ISSUE THREAD

**Issue by ericnakagawa** [facebook/Docusaurus#101]
Feature Request:

Allow defining a permalink in post and doc markdown frontmatter.

**Comment by steveklabnik** [comment]
What does "permalink" mean in this context? If it's setting the URL that a page is configured at, like I described [over here](https://github.com/facebook/Docusaurus/issues/323#issuecomment-352915482), then a huge 👍 from me.

**Comment by JoelMarcey** [comment]
Hi @steveklabnik. Yes, like allowing the actual link that is navigated to be different than our default. 

e.g.,

```
permalink: /my/custom/link.html
```

**Comment by endiliey** [comment]
Implemented in `v2`

This is how it works

https://github.com/facebook/Docusaurus/blob/5d60739f7c9263f3f8142f27c85b0180eeb1f19f/v2/lib/load/docs/metadata.js#L117-L134

**Comment by zeluspudding** [comment]
How can one [map multiple URLs to the same page](https://stackoverflow.com/questions/74311247/multiple-urls-for-one-page)? I've tried multiple variations of the permalink syntax trying to get an endpoint like _example.com/productA/latest_ to also map to _example.com/productA/1.0.2_. Looking for this approach as we have multiple products we need to version and each should be discoverable via a @latest page.

**Comment by slorber** [comment]
> How can one [map multiple URLs to the same page](https://stackoverflow.com/questions/74311247/multiple-urls-for-one-page)? I've tried multiple variations of the permalink syntax trying to get an endpoint like _example.com/productA/latest_ to also map to _example.com/productA/1.0.2_ but I can't seem to make it work. Looking for this approach as we have multiple products we need to version and each should be discoverable via a /latest page.

Docusaurus is only responsible for building static html files.

It is possible to use our client-side redirect plugin to emit extra files at `build/productA/latest/xyz.html` that would redirect with JS to `/productA/1.0.2/xyz`, but it is not ideal and you should use your hosting provider features to setup a server redirect (much better for SEO than a client-side redirect).


--

If you do not want a redirect: you can also serve the same content from 2 separate locations. 

For example, if you run as a post-build step `cp -R build/productA/1.0.2 build/productA/latest`

Your pages will then become available at 2 distinct locations: `/productA/1.0.2/xyz` + `/productA/latest/xyz`

For both pages, the canonical SEO URL of your page will remain `/productA/1.0.2/xyz`

**Comment by zeluspudding** [comment]
Hey - just donated to the project, thank you so much for your help :)

[Here's a stackblitz link](https://stackblitz.com/edit/github-5c7qs5-jytg3l?file=docs/scious-search/latest.md) to my build. 
The build is hosted on Render.com at https://docs.scious.io/scious-search/latest
I have a redirect rule in Render to point `/scious-search/*` to `/scious-search/latest`. Redirects in Render [only kick in if no content is found](https://render.com/docs/redirects-rewrites#rule-matching-and-ordering) at the original path. 

With that, I would expect a visit to:
https://docs.scious.io/scious-search/this_doesnt_exist
to redirect to:
https://docs.scious.io/scious-search/latest
_which the browser does_ but causes the page to display:
```
This page isn’t working
site.com redirected you too many times.

ERR_TOO_MANY_REDIRECTS
```

That's why I think a partial disable of the internal 404 routing would resolve this... but not sure.
